docker_cli/exceptions.py

```diff
@@ -0,0 +1,14 @@
+from subprocess import CalledProcessError
+
+
+class CommandNotFoundOnImageError(CalledProcessError):
+    def __init__(self, prior_error: CalledProcessError, command_name: str):
+        super(CommandNotFoundOnImageError, self).__init__(
+            prior_error.returncode,
+            prior_error.cmd
```

Comment:
I removed the parent class from CommandNotFoundError but would still be interested to hear your thoughts on this

Reply:
I think both approaches are perfectly fine and your reasoning in each case makes sense to me. I have a slight preference for not inheriting from `CalledProcessError`, just because its simpler and requires a bit less code (and the [exception chaining](https://peps.python.org/pep-3134/) strategy already does a good job of preserving the details of the original `CalledProcessError`). But if you choose to go back to the previous approach, I think that's totally defensible and it'd be fine with me.